### PR TITLE
User should be able to reset password

### PR DESCRIPTION
**Pull request by Faustin250**
>When the user with the email exists

>When the token expired

>Use the generated token in URL to change the password

### CODE REVIEW

**Review by muhireheir — APPROVED** [review]
Approved

**Review by bizip — APPROVED** [review]

**Comment by Faustin250** [comment]
[Reset Password](https://trello.com/c/MxF6s3ol/22-reset-password)

**Review by mugisha1555 — APPROVED** [review]

File: src/routes/api/apiRoutes.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import express from 'express';
 import user from './users/userRoutes';
 
+const nodemailer = require('nodemailer');
+const mg = require('nodemailer-mailgun-transport');
+
```
Comment (nkalyesubula) [inline src/routes/api/apiRoutes.js]: Why do we need this code here?

**Review by nkalyesubula — CHANGES_REQUESTED** [review]
There is actually a duplication of html template for sending email notifications i.e for password reset & email activation. Therefore lets have on HTML template that takes in email, the message to be sent, and first name of user you're sending it to. Also let's look out for a better looking email..not just a plain text

**Pull request merged**